src/main/java/htsjdk/variant/vcf/VCFCompoundHeaderLine.java

```diff
@@ -151,78 +189,126 @@ protected VCFCompoundHeaderLine(String name, VCFHeaderLineCount count, VCFHeader
     /**
      * create a VCF format header line
      *
-     * @param line   the header line
-     * @param version      the VCF header version
-     * @param lineType     the header line type
-     *
+     * @param line     the header line
+     * @param version  the VCF header version
+     * @param lineType the header line type
      */
     protected VCFCompoundHeaderLine(String line, VCFHeaderVersion version, SupportedHeaderLineType lineType) {
         super(lineType.toString(), "");
 
         final ArrayList<String> expectedTags = new ArrayList(Arrays.asList("ID", "Number", "Type", "Description"));
-        if (version.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_2))
-            expectedTags.add("Version");
-        final Map<String, String> mapping = VCFHeaderLineTranslator.parseLine(version, line, expectedTags);
+        final List<String> optionalTags;
+        if (version.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_2)) {
+            optionalTags = Arrays.asList("Source", "Version");
+        }
+        else {
+            optionalTags = Collections.emptyList();
+        }
+        final Map<String, String> mapping = VCFHeaderLineTranslator.parseLine(version,
+                                                                              line,
+                                                                              expectedTags,
+                                                                              optionalTags);
         name = mapping.get("ID");
         count = -1;
         final String numberStr = mapping.get("Number");
         if (numberStr.equals(VCFConstants.PER_ALTERNATE_COUNT)) {
             countType = VCFHeaderLineCount.A;
-        } else if (numberStr.equals(VCFConstants.PER_ALLELE_COUNT)) {
+        }
+        else if (numberStr.equals(VCFConstants.PER_ALLELE_COUNT)) {
             countType = VCFHeaderLineCount.R;
-        } else if (numberStr.equals(VCFConstants.PER_GENOTYPE_COUNT)) {
+        }
+        else if (numberStr.equals(VCFConstants.PER_GENOTYPE_COUNT)) {
             countType = VCFHeaderLineCount.G;
-        } else if ((version.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_0) && numberStr.equals(VCFConstants.UNBOUNDED_ENCODING_v4)) ||
-                (!version.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_0) && numberStr.equals(VCFConstants.UNBOUNDED_ENCODING_v3))) {
+        }
+        else if ((
+                         version.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_0) &&
+                         numberStr.equals(VCFConstants.UNBOUNDED_ENCODING_v4)
+                 ) ||
+                 (
+                         !version.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_0) &&
+                         numberStr.equals(VCFConstants.UNBOUNDED_ENCODING_v3)
+                 )) {
```

Comment:
Gah, IntelliJ. If you published code style settings that I can load into IntelliJ to prevent these issues, that would be a huge time saver + you would get cleaner code.

Reply:
I think using the default intellij settings will work, with the exception of `case` statements, which are indented (vs K&R style where they align with the `switch`). Generally the code uses oracle/sun conventions.

Reply:
> I was reverting things like Map<String, String> to Map<String,String>, there are missing { }s around if statements, max line length is not honored, things are indented strangely in places... If you simply run "reformat file" and you get dozens of changes, the development becomes quite cumbersome.

Instead of reformatting an entire file, select the new/modified code and use `Code/Reformat Code`. If existing code isn't formatted correctly but isn't being modified by your PR, you can leave it for now.

> I would recommend taking default IntelliJ settings, actually reformat everything to that (you will be surprised to see all the changes), publish the settings (you can export them as a .jar), and then just keep things clean going forward.

It would be nice to fix the formatting of the entire code base, but as yet no-one has taken that work on. In the meantime, we can avoid adding new code that doesn't conform to the conventions, and fix code along the way as it's modified.

> As a reviewer, do not waste time policing coding style, we have tools for that. Just enable people to use those tools.

I agree that it would be better if there's less code style discussions in code reviews. Submitters can help by making sure that any new/modified code follows the code style..
  
While code reviews are a valuable tool for identifying problems, sharing knowledge, etc, they are also a tool to ensure that the code stays in a form that's easy for anyone to maintain. Moving to and/or preserving a consistent coding style is part of that process.

Regarding tooling there have been some suggestions but again, no-one has done the work to enable, say, checkstyle on the codebase. Using something like checkstyle would help a lot but requires an investment to get working and integrated in the build.